test/jdk/sun/security/pkcs11/ec/TestECDSA.java

```diff
@@ -167,6 +180,11 @@ public void main(Provider provider) throws Exception {
         }
         test(provider, pub521, priv521, sig521);
 
+        if (provider.getName().equalsIgnoreCase("SunEC")) {
```

Comment:
Please add a comment explaining why this test is conditional (or make it unconditional).

Reply:
Comment added to this code explaining the if statement added:
```
        // This test is known to be executed in two ways:
        // 1. Direct execution of this test for testing sun.security.pkcs11
        // functionality. Skip brainpoolP512r1 tests in this case since the PKCS11
        // provider does not support them.
        // 2. Running the testcase in sun/security/ec. Expect brainpoolP512r1
        // curve to be present execute the brainpoolP512r1 specific test.
```